src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringRunListener.java

```diff
@@ -124,15 +127,15 @@ public void _onInitialize(Run run) {
 
         // TODO move this to a pluggable span enrichment API with implementations for different observability backends
         // Regarding the value `unknown`, see https://github.com/jenkinsci/opentelemetry-plugin/issues/51
-        rootSpanBuilder
-                .setAttribute(JenkinsOtelSemanticAttributes.ELASTIC_TRANSACTION_TYPE, "unknown");
 
-        rootSpanBuilder
-                .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_ID, rootSpanName)
-                .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_NAME, run.getParent().getFullDisplayName())
-                .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_URL, runUrl)
-                .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_NUMBER, (long) run.getNumber())
-                .setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_TYPE, OtelUtils.getProjectType(run));
+        AttributesBuilder attrBuilders = Attributes.builder()
```

Comment:
Use `AttributesBuilder` to be able to consume it later on to populate the attributes in the `MonitoringAction`